core/src/test/java/org/apache/iceberg/TestSequenceNumberForV2Table.java

```diff
@@ -0,0 +1,299 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestSequenceNumberForV2Table extends TableTestBase {
+
+  public TestSequenceNumberForV2Table() {
+    super(2);
+  }
+
+  @Test
+  public void testSequenceNumberForFastAppend() throws IOException {
+    table.newFastAppend().appendFile(FILE_A).commit();
+    Assert.assertEquals(1, table.currentSnapshot().sequenceNumber());
+    ManifestFile manifestFile = table.currentSnapshot().manifests().get(0);
+    Assert.assertEquals(1, manifestFile.sequenceNumber());
+
+    table.newFastAppend().appendFile(FILE_B).commit();
+    Assert.assertEquals(2, table.currentSnapshot().sequenceNumber());
+    manifestFile = table.currentSnapshot().manifests().stream()
+        .filter(manifest -> manifest.snapshotId() == table.currentSnapshot().snapshotId())
+        .collect(Collectors.toList()).get(0);
+    Assert.assertEquals(2, manifestFile.sequenceNumber());
+
+    manifestFile = writeManifest(FILE_C, FILE_D);
+    table.newFastAppend().appendManifest(manifestFile).commit();
+    Assert.assertEquals(3, table.currentSnapshot().sequenceNumber());
+
+    manifestFile = table.currentSnapshot().manifests().stream()
+        .filter(manifest -> manifest.snapshotId() == table.currentSnapshot().snapshotId())
+        .collect(Collectors.toList()).get(0);
+    Assert.assertEquals(3, manifestFile.sequenceNumber());
+
+    for (ManifestEntry entry : ManifestFiles.read(manifestFile,
+        table.io(), table.ops().current().specsById()).entries()) {
+      if (entry.file().path().equals(FILE_C.path()) || entry.file().path().equals(FILE_D.path())) {
+        Assert.assertEquals(3, entry.sequenceNumber().longValue());
+      }
+    }
+  }
+
+  @Test
+  public void testSequenceNumberForMergeAppend() throws IOException {
```

Comment:
There should be two cases for this test. One with a default merge setting (no merging) and one with the min merge count changed.

Reply:
Done